azurerm/data_source_api_management.go

```diff
@@ -0,0 +1,288 @@
+package azurerm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceApiManagementService() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceApiManagementRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"location": locationForDataSourceSchema(),
+
+			"publisher_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"publisher_email": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"sku": apiManagementDataSourceSkuSchema(),
```

Comment:
Sure, but just to make sure I understand your preference, since I see you have not mentioned `apiManagementDataSourceCertificateInfoSchema`. What I've done is create funcs for schemas that are used multiple locations. Both `apiManagementDataSourceSkuSchema` and `apiManagementDataSourceCertificateInfoSchema` is used twice in the overall data source schema.
  
If this is not a good practice here (or in general 😃 ) I'll inline them.

Reply:
tbh in this case I think we could probably remove the second SKU in the data source, since we've also removed it from the resource (which is why I originally posted this)

(in general we tend to follow the Go guidelines of duplicating code the first time, and then refactoring it into a method - but it's mostly a gut feeling based on where something's likely to diverge, in this case I'd argue it probably wouldn't since it's the same object in both cases)